common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```diff
@@ -257,25 +281,36 @@ public Object run() {
             throws ClassNotFoundException {
         try {
             return loader.loadClass(helper.getName());
-        } catch (ClassNotFoundException e) {
-            // The helper class is NOT found in target ClassLoader, we have to define the helper class.
-            final byte[] classBinary = classToByteArray(helper);
-            return AccessController.doPrivileged(new PrivilegedAction<Class<?>>() {
-                @Override
-                public Class<?> run() {
-                    try {
-                        // Define the helper class in the target ClassLoader,
-                        //  then we can call the helper to load the native library.
-                        Method defineClass = ClassLoader.class.getDeclaredMethod("defineClass", String.class,
-                                byte[].class, int.class, int.class);
-                        defineClass.setAccessible(true);
-                        return (Class<?>) defineClass.invoke(loader, helper.getName(), classBinary, 0,
-                                classBinary.length);
-                    } catch (Exception e) {
-                        throw new IllegalStateException("Define class failed!", e);
+        } catch (ClassNotFoundException e1) {
+            try {
+                // The helper class is NOT found in target ClassLoader, we have to define the helper class.
+                final byte[] classBinary = classToByteArray(helper);
+                return AccessController.doPrivileged(new PrivilegedAction<Class<?>>() {
+                    @Override
+                    public Class<?> run() {
+                        try {
+                            // Define the helper class in the target ClassLoader,
+                            //  then we can call the helper to load the native library.
+                            Method defineClass = ClassLoader.class.getDeclaredMethod("defineClass", String.class,
```

Comment:
Also, this looks like a possible bug.  Shouldn't `loader` be used to lookup `defineClass`, not `ClassLoader.class` ?

cc: @ejona86

Reply:
good question...

Reply:
Since `loader` is known to be an instanceof `ClassLoader` using `ClassLoader.class` here is fine. It stands out as strange because the reflection is to avoid accessibility, not because the class is unknown at compile time. So it's good and fine as-is, just most other reflection usages couldn't do this.